src/context.rs

```diff
@@ -298,10 +298,12 @@ fn update<S: ScheduleLabel>(
         };
 
         let gamepad = context.get::<GamepadDevice>().copied().unwrap_or_default();
-        let Some(context_actions) = instance.actions_mut(&mut context) else {
+        let Some(mut context_actions) = instance.actions_mut(&mut context) else {
             continue;
         };
 
+        // We only sort actions, without actually modifying elements the collection.
+        let context_actions = context_actions.bypass_change_detection();
```

Comment:
I would instead guard this by a `.is_sorted` call or the like.

Reply:
This might be a good idea :thinking: Let me try.